src/m/m3d/m_shadow.cpp

```diff
@@ -0,0 +1,604 @@
+#include <m/m3d/m_shadow.h>
+#include <nw4r/g3d/g3d_calcview.h>
+#include <nw4r/g3d/g3d_draw.h>
+#include <nw4r/g3d/g3d_draw1mat1shp.h>
+#include <nw4r/g3d/g3d_resmat.h>
+#include <nw4r/g3d/g3d_resmdl.h>
+#include <nw4r/g3d/g3d_resshp.h>
+#include <nw4r/g3d/g3d_scnmdl.h>
+#include <nw4r/g3d/g3d_scnmdlsmpl.h>
+#include <nw4r/g3d/g3d_state.h>
+
+// All of this is completely made up, as we don't have symbols for this TU
+// (contrary to the rest of m3d and most of nw4r::g3d)
+
+#define FRM_HEAP_STATE 0x74616730
```

Comment:
this is "tag0", I believe hex representation isn't needed for 4-byte strings (i want to safely remove the multi-byte warning lol)